view/frontend/templates/subscription_freq_selector_product_page.phtml

```diff
@@ -0,0 +1,183 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2022 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+// return if PDP FS option is disabled or we aren't on product page
+if (!$block->isBoltProductPageForFrequencySelector()) { return;
+}
+?>
+
+
+<!--
+    We will potentially embed the frequency selector into this selector.
+    If product is subscribable - it will load the frequency options
+    If not - it will load nothing (empty div)
+ -->
+<div class="bolt-freq-selector"></div>
+
+<script>
+    require([
+        'jquery',
+        'Magento_Customer/js/model/authentication-popup',
+        'Magento_Customer/js/customer-data',
+        'Bolt_Boltpay/js/utils/when-defined',
+        'mage/validation/validation',
+        'mage/cookies',
+        'jquery/jquery.cookie',
+        'domReady!'
+    ], function ($, authenticationPopup, customerData, whenDefined) {
+
+        var callbacks = {
+            /**
+             * This callback is triggered by the frequency selector's onChange
+             * User has selected a different frequency
+             * 
+             * We update the localStorage value that is accessible when shopper checks out
+             * LocalStorage value is a dictionary with
+             * - key as merchant product id ("reference" here)
+             * - value is the selected frequency (value, unit eg: 2, 'weeks')
+             */
+            onSubscriptionChanged: function(reference, frequency) {
+                console.log("subscription frequency changed! ", reference, frequency);
+                const frequencies = JSON.parse(localStorage.getItem("BOLT_SUBSCRIPTION_PDP")) ?? {}
+                if (!frequency) {
+                    delete frequencies[reference]
+                } else {
+                    const { unit, value } = frequency
+                    frequencies[reference] = {
+                        unit,
+                        value
+                    }
+                }
+                localStorage.setItem("BOLT_SUBSCRIPTION_PDP", JSON.stringify(frequencies))
+            }
+        }
+
+        /**
+         * Below are util funcs used to retrieve full product info
+         * referenced from button_product_page.phtml
+         */
+        var getQty = function () {
+            var quantity = Number($('#qty').val());
+            return quantity > 0 ? quantity : 1;
+        }
+
+        var getGroupedProductChildQty = function () {
+            return Number($('input[name="super_group['+productId+']').val());
+        }
+
+        var getItemsData = function () {
```

Comment:
All of these came from https://github.com//BoltApp/bolt-magento2/blob/master/view/frontend/templates/button_product_page.phtml#L211 - all used to get the items on this page. We need the items on the BoltConnect side to identify if the product(s) are subscribable.